typescript/src/entities/products.ts

```diff
@@ -1,4 +1,15 @@
 import { Entity } from '../entity';
 
 export class Product extends Entity {
+  @Product.property("id")
+  private _id?: number;
+
+  get id(): number | undefined {
+    return this._id;
+  }
+
+  set id(newId: number | undefined) {
+    this._id = newId;
+    this._isDirty = true;
+  }
```

Comment:
11 lines of code to introduce one new attribute is a little bit wordy, but i think it's about the best i can do at the moment

Reply:
Will this be the case for every attribute?

Reply:
yep!

there's some vaguely proposed things for improved decorators in typescript which would make it easier, but they don't exist yet, and might never.

there's a possibility of applying a pre-processor, including the "transformer" thing kind of build into tsc, but, i had a look at that and i thought that it might be more trouble than its worth at the moment. not enough people use it so the documentation and interface is real bad.